src/magic_folder/uploader.py

```diff
@@ -247,6 +253,10 @@ def upload_local_snapshots(self):
         # get the mangled paths for the LocalSnapshot objects in the db
         localsnapshot_names = self._config.get_all_localsnapshot_paths()
 
+        # update our status if we have nothing to do
+        if not len(localsnapshot_names):
+            self._status.upload_stopped()
```

Comment:
It occurs to me this only says that *this* folder had no snapshots.

Reply:
Hmm .. yeah.

For me, 90% of the point of this PR is "get the infrastructure in place". Maybe GridSync actually wants syncing status per-folder (for example) which would make this easier to get right anyway.

Reply:
I guess either way, the "status" thing will need to track per-folder activity (and possibly the serialized version of that just says "sync" or "not-sync" globally) so points to changing the `upload_stopped()` / `upload_started()` APIs to take a folder-name.

Reply:
okay, I changed that (to track per-folder, but still just output a single bool).